src/main/java/org/pawaneuler/Generator/TransactionLogGenerator/RandomGenerator.java

```diff
@@ -46,53 +46,44 @@ public Random getRandomizer() {
     }
 
     /**
+     * to get random number of String
      * 
      * @return
      */
     public String[] randomStrings() {
         int randomNumOfList = randomizer.nextInt(maxVariety);
-        String[] randomString = new String[randomNumOfList];
-        ArrayList<String> random = randomList(randomNumOfList);
-        
-        for (int i = 0; i < randomNumOfList; i++) {
-            randomString[i] = random.get(i);
-        }
-        return randomString;
+        return getStrings(productVariety, randomNumOfList);
     }
 
     /**
      * 
-     * @return random list of product with no duplicate and sorted
+     * @param numOfList number of list
+     * @return random list with the length that passed in poarameter
      */
-    public ArrayList<String> randomList(int randomNumOfList) {
-        ArrayList<String> random = new ArrayList<String>();
-        int randomIndex;
-        while (random.size() < randomNumOfList) {
-            randomIndex = randomizer.nextInt(maxVariety);
-            
-            //to make sure there's no duplicate 
-            random.remove(productVariety.get(randomIndex));
-            random.add(productVariety.get(randomIndex));
-            
-            Collections.sort(random);
+    public String[] getStrings(String[] strings, int numOfList) {
```

Comment:
Better function name this way?
```suggestion
    public String[] getNStringsFrom(int numOfList, String[] strings) {
```